eos_data_distribution/ndn/test.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/python
+# -*- Mode:python; coding: utf-8; c-file-style:"gnu"; indent-tabs-mode:nil -*- */
+#
+# Copyright (C) 2016 Endless Computers INC.
+# Author: Niv Sardi <xaiki@endlessm.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+# A copy of the GNU Lesser General Public License is in the file COPYING.
+
+from . import file
+from gi.repository import GLib
+import argparse
+
+import logging
+logger = logging.getLogger(__name__)
+
+def process_args(*extra):
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-n", "--name")
+    parser.add_argument("-o", "--output")
+    parser.add_argument("-v", action="count")
+    [parser.add_argument(a) for a in extra]
+
+    args = parser.parse_args()
+    if args.v == 0:
+        logging.basicConfig(level=logging.INFO)
+    else:
+        logging.basicConfig(level=logging.DEBUG)
+
+    return args
+
+def run_test(args, name):
+    loop = GLib.MainLoop()
+
+    if args.output:
+        consumer = file.FileConsumer(name, filename=args.output, auto=True)
```

Comment:
To automate these tests (`unittest`), you should be able to hardcode `name` and rework `FileConsumer` a bit so you can pass a `file`-like object (from `tempfile.NamedTemporaryFile`) to it instead of `filename`. That will allow the unit tests to use temporary files while avoiding race conditions with temporary filename generation.